src/components/ServiceDetails/AboutService/ServiceDemo/Purchase/ExpiredSession/GeneralAccountWallet/PaymentPopup/Details/index.js

```diff
@@ -13,31 +15,88 @@ import { useStyles } from "./styles";
 import snetValidator from "../../../../../../../../../utility/snetValidator";
 import { paymentGatewayConstraints } from "./validationConstraints";
 import AlertBox, { alertTypes } from "../../../../../../../../common/AlertBox";
-import { USDToAgi } from "../../../../../../../../../utility/PricingStrategy";
+import { USDToAgi, agiToCogs } from "../../../../../../../../../utility/PricingStrategy";
+import { orderTypes } from "../../";
+import { parseSignature } from "../../../../../../../../../utility/sdk";
+import { groupInfo } from "../../../../../../../../../Redux/reducers/ServiceDetailsReducer";
 
 export const paymentTypes = [{ value: "paypal", label: "Pay pal" }];
 
-const Details = ({ classes, initiatePayment, handleClose, channelInfo }) => {
+let web3;
+
+const Details = props => {
+  const {
+    classes,
+    initiatePayment,
+    handleClose,
+    channelInfo,
+    orderType,
+    userProvidedPrivateKey: privateKey,
+    groupInfo,
+  } = props;
+
   const [payType, setPayType] = useState("default");
   const [amount, setAmount] = useState("");
   const [alert, setAlert] = useState({});
   const [currency] = useState("USD");
 
+  useEffect(() => {
+    if (!web3) {
+      web3 = new Web3(process.env.REACT_APP_WEB3_PROVIDER, null, {});
+    }
+  });
+
   const handlePayTypeChange = event => {
     const { value } = event.target;
     if (value !== "default") {
       setPayType(value);
     }
   };
 
-  const handleContinue = () => {
+  const handleContinue = async () => {
     setAlert({});
     const isNotValid = snetValidator({ payType, amount }, paymentGatewayConstraints);
     if (isNotValid) {
       setAlert({ type: alertTypes.ERROR, message: isNotValid[0] });
       return;
     }
-    initiatePayment(payType, amount, currency, "AGI", USDToAgi(amount));
+    try {
+      const amountInAGI = USDToAgi(amount);
+      let base64Signature;
+      let address;
+      let currentBlockNumber;
+      if (orderType === orderTypes.CREATE_CHANNEL) {
+        const account = web3.eth.accounts.privateKeyToAccount(privateKey);
+        address = account.address;
+        web3.eth.accounts.wallet.add(account);
+        web3.eth.defaultAccount = address;
+        const recipient = groupInfo.payment.payment_address;
+        const hexGroupId = `0x${atob(groupInfo.group_id)}`;
+        const amountInCogs = agiToCogs(amountInAGI);
+        currentBlockNumber = await web3.eth.getBlockNumber();
+        // 1 block no is mined in 15 sec on average, setting expiration as 10 years
+        const expiration = currentBlockNumber + 10 * 365 * 24 * 60 * 4;
```

Comment:
`10 * 365 * 24 * 60 * 4` could be extracted into a constant, something like `ten_year_block_offset`.

Reply:
https://github.com/singnet/snet-dapp/pull/414/commits/d05838a1fae76427d3dcd354c33472f401d54c6e